new_study_assistant.py

```diff
@@ -0,0 +1,624 @@
+    
+#check matching attributes (via name only)
+exact_matching_attributes = np.intersect1d(test_study_attribute_names, list(study_data_combined))
+non_matching_attributes = np.setdiff1d(test_study_attribute_names, list(study_data_combined))
+possible_matches = find_attribute_name_matches(study_data_combined, non_matching_attributes)
+
+#check matching attributes based on data values
+if api_flag:
+    patient_data = get_clinical_data_values(all_cBioPortalStudies, attribute_type="PATIENT")
+    patient_data_processed = process_clinical_data(patient_data, study_to_drop)
+    sample_data = get_clinical_data_values(all_cBioPortalStudies, attribute_type="SAMPLE")
+    sample_data_processed = process_clinical_data(sample_data, study_to_drop)
+    combined_PS_attribute_data = pd.concat([patient_data_processed, sample_data_processed], axis=0).fillna(value=0)
+else:
+    combined_PS_attribute_data = process_clinical_data(attribute_value_data, study_to_drop)
+combined_attribute_values = pd.concat([combined_PS_attribute_data, new_study_clinical_attribute_values], axis=0).fillna(value=0)
+
+#get clusters and plot dendrogram
+clusters = get_clusters(combined_attribute_values)
+
+#process matches
+processed_name_matches, no_name_matches = process_levenshtein_matches(exact_matching_attributes, 
+                                                                      possible_matches,
+                                                                      non_matching_attributes)
+processed_value_matches = process_cluster_matches(clusters)
+
+#combine matches into one dataframe
+exact_matches_df = pd.DataFrame(np.column_stack([exact_matching_attributes, exact_matching_attributes]), 
+                                columns=['New study attribute', 'Possible matches'])
+name_matches_df = pd.DataFrame(processed_name_matches, columns=['New study attribute', 'Possible matches'])
+value_matches_df = pd.DataFrame(processed_value_matches, columns=['New study attribute', 'Possible matches'])
+no_name_or_value_matches = get_unique_attributes(no_name_matches, value_matches_df)
+no_matches_df = pd.DataFrame(np.asarray(([no_name_or_value_matches, ['No matches found']*len(no_name_or_value_matches)])).T,
+                             columns=['New study attribute', 'Possible matches'])
+all_matches_df = combine_matches(exact_matches_df, name_matches_df, value_matches_df, no_matches_df)
+
+predicted_attribute_types=None
+if api_flag:
+    predicted_attribute_types = attribute_type_prediction(exact_matches_df, name_matches_df, value_matches_df, studies)
+
+if new_study_path != None:
+    test_study = new_study_path
+#print all_matches_df
+if latex_flag:
+    generate_latex_report(all_matches_df, predicted_attribute_types, test_study)
+    print "Results written to report.pdf"
+else:
+    generate_html_report(all_matches_df, predicted_attribute_types, test_study)
+    print "Results written to report.html"
+    #print "report for study: " + test_study
```

Comment:
if these are not needed, I would remove the comments. If they are still useful (maybe for debugging?) then I would add a --debug command line argument and print them conditionally in debug mode.